lib/jormungandr/src/Cardano/Wallet/Jormungandr/Primitive/Types.hs

```diff
@@ -28,7 +30,13 @@ import GHC.Generics
     ( Generic )
 
 data Tx = Tx
-    { inputs
+    { txid
+        :: Hash "Tx"
+        -- ^ Jörmungandr computes transaction id by hashing the full content of
+        -- the transaction, which includes witnesses. Therefore, we need either
+        -- to keep track of the witnesses to be able to re-compute the tx id
+        -- every time, or, simply keep track of the id itself.
```

Comment:
hope in the next version of Jormungandr will not change how  they compute transaction id again ...

Reply:
I think that decision is rather final (they basically now compute only "fragment ids", so the idea is even more holistic than a transaction id. A transaction is one type of "fragment", but there are many other types, and it allows for an open extension). 

Yet, they might change some other stuff that may not be trivial to update either :man_shrugging: